src/jit/assertionprop.cpp

```diff
@@ -3058,9 +3075,9 @@ GenTree* Compiler::optAssertionPropGlobal_RelOp(ASSERT_VALARG_TP assertions, Gen
     bool allowReverse = true;
 
     // If the assertion involves "op2" and it is a constant, then check if "op1" also has a constant value.
-    if (vnStore->IsVNConstant(op2->gtVNPair.GetConservative()))
+    ValueNum vnCns = vnStore->VNNormVal(op2->gtVNPair.GetConservative());
```

Comment:
There are so many of these VNNormVal that perhaps we should add a shortcut to `Compiler` - `ValueNum Compiler::vnConservativeNormVal(GenTree*);`

Reply:
Yes, I am considering adding  heper functions:

vnConservativeNormVal
vnLiberalNormVal